miro_preprocessor/xml_to_json_converter/src/run.py

```diff
@@ -47,6 +48,8 @@ def main(bucket, src_key, dst_key, js_path="json"):
 
     s3 = boto3.client('s3')
 
+    image_data_count = 1
```

Comment:
You don’t need to manage the loop yourself.

Replace L54 with:

```python
for count, img in enumerate(image_data, start=1):
```

See https://docs.python.org/3/library/functions.html#enumerate

Reply:
Will that work on a generator?

Reply:
Yes!

```python
def infinite():
    x = 1
    while True:
        x *= 2
        yield x

for i, x in enumerate(infinite()):
    print(i, x)
    if i > 10:
        break
```